testsuite/backend/spicedb.py

```diff
@@ -0,0 +1,66 @@
+"""SpiceDB implementation of Backend"""
+
+from testsuite.config import settings
+from testsuite.backend import Backend
+from testsuite.kubernetes import Selector
+from testsuite.kubernetes.client import KubernetesClient
+from testsuite.kubernetes.deployment import Deployment
+from testsuite.kubernetes.service import Service, ServicePort
+
+
+class SpicedbBackend(Backend):
+    """SpiceDB deployed in Kubernetes."""
+
+    def __init__(self, cluster: KubernetesClient, name, label, image, preshared_key="secret", replicas=1) -> None:
+        super().__init__(cluster, name, label)
+        self.replicas = replicas
+        self.image = image
+        self.preshared_key = preshared_key
+        self.secret = None
+
+    @property
+    def grpc_port(self):
+        """Returns the gRPC port number"""
+        return self.service.get_port("grpc").port
+
+    @property
+    def grpc_url(self):
+        """Returns the external gRPC URL for SpiceDB"""
+        return f"{self.service.external_ip}:{self.grpc_port}"
+
+    def commit(self):
+        match_labels = {"app": self.label, "deployment": self.name}
+
+        self.deployment = Deployment.create_instance(
+            self.cluster,
+            self.name,
+            container_name="spicedb",
+            image=self.image,
+            ports={"grpc": 50051},
+            selector=Selector(matchLabels=match_labels),
+            labels={"app": self.label},
+            command_args=[
+                "serve",
+                "--grpc-preshared-key",
+                self.preshared_key,
+            ],
+        )
+        self.deployment.commit()
+        self.deployment.wait_for_ready()
+
+        self.service = Service.create_instance(
+            self.cluster,
+            self.name,
+            selector=match_labels,
+            ports=[
+                ServicePort(name="grpc", port=50051, targetPort="grpc"),
+            ],
+            labels={"app": self.label},
+            service_type="LoadBalancer",
```

Comment:
Is the service used outside the cluster like multicluster test? If not, we maybe don't need to use LoadBalancer type here. This could also apply to some services in tools. Needs investigation. 
Especially if you want to choose hard-coded secret.